### PR TITLE
x892 Highlight barcode on history page

### LINKED ISSUE #196: x892 - Highlight barcode in history page
**User story**
As a PO (Minal) I would like the barcode in the history page to be highlighted to make it quick and easy to find the barcode in the history

**Who are the primary contacts for this story**
Minal

**Acceptance criteria**
To be considered successful the solution must allow:
- [ ] if searching by barcode - highlight the relevant barcode in the history 
- [ ] Make text background yellow for search term barcode in the result columns

### PR DESCRIPTION
**Pull request by BenTopping**
Closes #196 

- Added background colour to matching barcodes on history page

### CODE REVIEW

File: src/components/history/History.tsx [hunk 1]
```diff
@@ -24,79 +24,92 @@ export default function History(props: HistoryProps) {
...
+      },
+      {
+        Header: 'Source',
+        accessor: 'sourceBarcode',
+        Cell: (props: Cell<HistoryTableEntry>) => {
+          const barcode = props.row.original.sourceBarcode;
+          let style = undefined;
+          if (historyProps.kind === 'labwareBarcode' && barcode === historyProps.value) {
+            style = { backgroundColor: 'yellow' };
+          }
```
Comment (seenanair) [inline src/components/history/History.tsx:53]: Just a thought - may be a 'padding' as well to make the highlight bigger/obvious. I don't know whether it looks good, but may be worth a try

**Review by seenanair — COMMENTED** [review]
Looks good 👍 
Do we need a test case for this highlight?

File: src/components/history/History.tsx (hunk 1 above)
Comment (BenTopping) [reply src/components/history/History.tsx:54]: Tested with padding and it makes the positioning different a bit different to the non-highlighted barcodes so it looks a bit weird.

**Review by khelwood — COMMENTED** [review]
Looks good, but the linter has some suggestions that look reasonable

**Review by khelwood — APPROVED** [review]

**Comment by BenTopping** [comment]
> Looks good, but the linter has some suggestions that look reasonable

Yep, fixed those lint suggestions in another PR https://github.com/sanger/stan-client/pull/183

**Pull request merged**